From 0b54cc8fa281a7a38a5928468ee112a3ae5ddb86 Mon Sep 17 00:00:00 2001
From: Vasyl Yushchyshen <vasyl.yushchyshen@ti.com>
Date: Wed, 18 Feb 2015 09:20:09 -0800
Subject: [PATCH 079/111] Remove heapmask from ion_client_create() call
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

New ION API does not need heap mask param, so it is deleted from the
calling funcs.

Change-Id: If7bfe91ec23e33063bb7465333c6f7d1c8381b66
Signed-off-by: Vasyl Yushchyshen <vasyl.yushchyshen@globallogic.com>
[hashcode0f@gmail.com: added fixes in omaprpc and smc where ion_client_create
was used.]
Signed-off-by: Hashcode <hashcode0f@gmail.com>
Signed-off-by: D. Andrei Măceș <dmaces@nd.edu>
---
 drivers/rpmsg/omaprpc/omap_rpc.c                   |  5 +----
 drivers/rpmsg/rpmsg_omx.c                          | 10 +---------
 drivers/video/omap2/omaplfb/omaplfb_displayclass.c |  6 +-----
 security/smc/tf_conn.c                             |  4 +---
 security/smc/tf_device.c                           |  4 +---
 5 files changed, 5 insertions(+), 24 deletions(-)

diff --git a/drivers/rpmsg/omaprpc/omap_rpc.c b/drivers/rpmsg/omaprpc/omap_rpc.c
index e67b478..ec2a150 100644
--- a/drivers/rpmsg/omaprpc/omap_rpc.c
+++ b/drivers/rpmsg/omaprpc/omap_rpc.c
@@ -498,10 +498,7 @@ static int omaprpc_open(struct inode *inode, struct file *filp)
 	}
 #if defined(OMAPRPC_USE_ION)
 	/* get a handle to the ion client for RPC buffers */
-	rpc->ion_client = ion_client_create(omap_ion_device,
-					    (1 << ION_HEAP_TYPE_CARVEOUT) |
-					    (1 << OMAP_ION_HEAP_TYPE_TILER),
-					    "rpmsg-rpc");
+	rpc->ion_client = ion_client_create(omap_ion_device, "rpmsg-rpc");
 #endif
 
 	/* remember rpc in filp's private data */
diff --git a/drivers/rpmsg/rpmsg_omx.c b/drivers/rpmsg/rpmsg_omx.c
index 425085a..0d53007 100644
--- a/drivers/rpmsg/rpmsg_omx.c
+++ b/drivers/rpmsg/rpmsg_omx.c
@@ -693,15 +693,7 @@ static int rpmsg_omx_open(struct inode *inode, struct file *filp)
 		return -ENOMEM;
 	}
 #ifdef CONFIG_ION_OMAP
-	omx->ion_client = ion_client_create(omap_ion_device,
-					    (1 << ION_HEAP_TYPE_CARVEOUT) |
-#ifdef CONFIG_MACH_TUNA
-					    (1 << OMAP_ION_HEAP_TYPE_TILER),
-#else
-					    (1 << OMAP_ION_HEAP_TYPE_TILER) |
-					    (1 << ION_HEAP_TYPE_SYSTEM),
-#endif
-					    "rpmsg-omx");
+	omx->ion_client = ion_client_create(omap_ion_device, "rpmsg-omx");
 #endif
 
 	init_completion(&omx->reply_arrived);
diff --git a/drivers/video/omap2/omaplfb/omaplfb_displayclass.c b/drivers/video/omap2/omaplfb/omaplfb_displayclass.c
index 3dba6f2..39e41c7 100644
--- a/drivers/video/omap2/omaplfb/omaplfb_displayclass.c
+++ b/drivers/video/omap2/omaplfb/omaplfb_displayclass.c
@@ -1192,11 +1192,7 @@ static OMAPLFB_ERROR OMAPLFBInitIonOmap(OMAPLFB_DEVINFO *psDevInfo,
 	struct sgx_omaplfb_config *psFBPlatConfig = GetFBPlatConfig(uiFBDevID);
 
 #if defined(CONFIG_ION_OMAP)
-        gpsIONClient = ion_client_create(omap_ion_device,
-                        1 << ION_HEAP_TYPE_CARVEOUT |
-                        1 << OMAP_ION_HEAP_TYPE_TILER |
-                        1 << ION_HEAP_TYPE_SYSTEM,
-                        "omaplfb");
+        gpsIONClient = ion_client_create(omap_ion_device, "omaplfb");
         if (IS_ERR_OR_NULL(gpsIONClient))
         {
 		printk(KERN_ERR DRIVER_PREFIX
diff --git a/security/smc/tf_conn.c b/security/smc/tf_conn.c
index 3ae505c..7a33e0e 100644
--- a/security/smc/tf_conn.c
+++ b/security/smc/tf_conn.c
@@ -1174,9 +1174,7 @@ int tf_invoke_client_command(
 
 			if (connection->ion_client == NULL) {
 				connection->ion_client = ion_client_create(
-					zebra_ion_device,
-					(1 << ION_HEAP_TYPE_CARVEOUT),
-					"tf");
+					zebra_ion_device, "tf");
 			}
 			if (connection->ion_client == NULL) {
 				dprintk(KERN_ERR "%s(%p): "
diff --git a/security/smc/tf_device.c b/security/smc/tf_device.c
index a88631e..7e76b2c 100644
--- a/security/smc/tf_device.c
+++ b/security/smc/tf_device.c
@@ -578,9 +578,7 @@ static long tf_device_ioctl(struct file *file, unsigned int ioctl_num,
 		/* Initialize ION connection */
 		if (connection->ion_client == NULL) {
 			connection->ion_client = ion_client_create(
-						zebra_ion_device,
-						(1 << ION_HEAP_TYPE_CARVEOUT),
-						"tf");
+						zebra_ion_device, "tf");
 		}
 
 		if (connection->ion_client == NULL) {
-- 
2.7.4

